src/client/app/actions/groups.ts

```diff
@@ -77,290 +97,129 @@ export function fetchGroupChildrenIfNeeded(groupID: number) {
-			} else {
-				showErrorNotification(translate('failed.to.create.group'));
-			}
+			// Update the groups state from the database on a successful call
+			// In the future, getting rid of this database fetch and updating the store on a successful API call would make the page faster
+			// However, since the database currently assigns the id to the GroupData and it is not returned we do the get.
+			// We also need to get the child meters/groups of the new group. We can just fetch this one group but instead get all the groups since easier and this
+			// is not a common operation. We must wait for the new group state so its substate for children can be set.
+			dispatch(fetchGroupsDetails()).then(() => dispatch(fetchAllGroupChildren()));
+			showSuccessNotification(translate('group.successfully.create.group'));
+		} catch (err) {
+			// Failure! ):
+			// TODO Better way than popup with React but want to stay so user can read/copy.
+			window.alert(translate('group.failed.to.edit.group') + '"' + err.response.data as string + '"');
+			// Clear our changes from to the submitting meters state
+			// We must do this in case fetch failed to keep the store in sync with the database
 		}
 	};
 }
 
-function submitGroupEdits(group: t.GroupData & t.GroupID): Thunk {
-	return async (dispatch: Dispatch) => {
-		dispatch(markGroupInEditingSubmitted());
-		try {
-			await groupsApi.edit(group);
-			dispatch(markGroupsOutdated());
-			dispatch(markOneGroupOutdated(group.id));
-			dispatch((dispatch2: Dispatch) => {
-				dispatch2(markGroupInEditingClean());
-			});
-		} catch (e) {
-			dispatch(markGroupInEditingNotSubmitted());
-			if (e.response.data.message && e.response.data.message === 'Cyclic group detected') {
-				showErrorNotification(translate('you.cannot.create.a.cyclic.group'));
-			} else {
-				showErrorNotification(translate('failed.to.edit.group'));
-			}
-		}
-	};
+export function confirmGroupEdits(editedGroup: t.GroupEditData): t.ConfirmEditedGroupAction {
+	return { type: ActionType.ConfirmEditedGroup, editedGroup };
 }
 
 /**
- * Checks if `groupInEditing` is dirty and not not submitted.
- * If this is the case, decides whether it is a new group
- * being created, or an old group being edited, and calls the
- * appropriate helper function to handle the request.
- * @returns {function(*, *)}
+ * Pushes group changes out to DB.
+ * @param group The group to update
+ * @param reload If true, the window is reloaded to reset everything on change
+ * @returns Function to do this for an action
  */
-export function submitGroupInEditingIfNeeded() {
-	return (dispatch: Dispatch, getState: GetState) => {
-		if (shouldSubmitGroupInEditing(getState())) {
-			const rawGroup = getState().groups.groupInEditing as t.GroupDefinition;
-			if (rawGroup === undefined) {
-				throw new Error('Unacceptable condition: state.groups.groupInEditing has no data.');
+export function submitGroupEdits(group: t.GroupEditData, reload: boolean = true): Thunk {
```

Comment:
This function contains a fair bit of todo/unused code, not sure if it needs to be cleaned up or if this is marked as part of the issue with reloading pages and will be fixed at a later date.

Reply:
Thanks for the reminder. As I feared, it took a little work to make sure state was properly used to update pages if a reload is not done but that is all fixed now. Thus, it only reloads when needed and dispatch is again used.